runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -157,15 +157,25 @@ void initializeSubgraphIOTensors(compiler::ILoweredGraph &lowered_graph,
 
 backend::BackendContexts
 createBackendContexts(compiler::ILoweredGraph &lgraph, bool linear_executor,
-                      std::shared_ptr<backend::custom::IKernelBuilder> custom_kernel_builder)
+                      std::shared_ptr<backend::custom::IKernelBuilder> custom_kernel_builder,
+                      const std::vector<std::string> &backend_list)
 {
   backend::BackendContexts contexts;
   auto &backend_manager = compiler::BackendManager::get();
 
   std::unordered_map<const backend::Backend *, backend::ContextData> context_data_map;
 
+  std::vector<const backend::Backend *> backends;
+  for (const auto &backend_name : backend_list)
+  {
+    auto backend = backend_manager.get(backend_name);
+    if (backend)
+      backends.push_back(backend);
+  }
+  backends.push_back(backend_manager.getBuiltin());
+
   // Generate partial graphs for each backend
-  for (auto &&backend : backend_manager.getAll())
+  for (auto &&backend : backends)
```

Comment:
As i understood correctly, the difference is here : 

#### Before
create backend from `backend_manager.getAll()`

#### After 
create backend from `CompilerOptions` config 


Is there a difference between `backend_manager.getAll()` and config lists?

Reply:
Yes, it can be different.
The scenario is like belows:

```mermaid
flowchart TD
   create_session --> id0["set_available_backends('train')"] -- "BACKNED config: ['cpu']\nbackend_manager: ['cpu']" --> train_prepare --> train --> id1["set_available_backends('cpu')"] -- "BACKNED config: ['train']\nbackend_manager: ['cpu', 'train']" --> prepare --> run --> close_session
```